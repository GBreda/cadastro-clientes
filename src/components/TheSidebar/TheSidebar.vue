<template>
  <div class="sidebar">
    <router-link
      v-for="(item, index) in routesList"
      :key="index"
      class="sidebar__link"
      :to="{ name: item.name }"
    >
      {{ item.meta.friendlyName }}
    </router-link>
  </div>
</template>

<script setup>
import { computed } from 'vue';
import { RouterLink, useRouter } from 'vue-router';

const router = useRouter();

const routesList = computed(() => router.options.routes);
</script>

<style lang="scss" scoped>
.sidebar {
  position: fixed;
  left: 0;
  padding: 2rem;
  background-color: $light-gray;
  box-shadow: 0px 0px 10px 0px rgba(0, 0, 0, 0.2);
  width: 30rem;
  min-height: 100vh;

  &__link {
    display: block;
    padding: 2rem;
    font-size: 1.4rem;
    border-left: 0.3rem solid transparent;

    &:hover {
      background-color: $gray;
    }

    &.router-link-active {
      border-color: $primary-color;
      color: $primary-color;
      font-weight: 600;
    }
  }
}
</style>
