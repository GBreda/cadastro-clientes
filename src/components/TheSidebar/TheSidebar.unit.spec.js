import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import { RouterLink, createRouter, createMemoryHistory } from 'vue-router';
import { nextTick } from 'vue';

import TheSidebar from './TheSidebar.vue';

const mockRouter = createRouter({
  history: createMemoryHistory(),
  routes: [
    {
      path: '/',
      name: 'customers-list',
      meta: {
        friendlyName: 'Lista de clientes'
      }
    },
    {
      path: '/lista-produtos',
      name: 'products-list',
      meta: {
        friendlyName: 'Lista de produtos'
      }
    },
    {
      path: '/cadastro-cliente',
      name: 'customer-registration',
      meta: {
        friendlyName: 'Cadastro de cliente'
      }
    },
    {
      path: '/cadastro-produto',
      name: 'product-registration',
      meta: {
        friendlyName: 'Cadastro de produto'
      }
    }
  ]
});

const mountTheSidebar = () => {
  return mount(TheSidebar, {
    global: {
      plugins: [mockRouter]
    },
    components: {
      RouterLink
    }
  });
};

describe('TheSidebar', () => {
  let wrapper;

  it('should mount the component', () => {
    wrapper = mountTheSidebar();

    expect(wrapper.vm).toBeDefined();
  });

  it('should match the snapshot', () => {
    wrapper = mountTheSidebar();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should have router-link-active class for active link', async () => {
    await mockRouter.push('/');

    await nextTick();

    const link = wrapper.findAll('.sidebar__link[href="/"]')[0];

    expect(link.classes()).toContain('router-link-active');
  });
});
