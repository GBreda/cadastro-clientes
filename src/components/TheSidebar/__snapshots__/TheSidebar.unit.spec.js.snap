// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TheSidebar > should match the snapshot 1`] = `
<div
  class="sidebar"
  data-v-28ca5493=""
>
  
  <a
    class="sidebar__link"
    data-v-28ca5493=""
    href="/"
  >
    Lista de clientes
  </a>
  <a
    class="sidebar__link"
    data-v-28ca5493=""
    href="/lista-produtos"
  >
    Lista de produtos
  </a>
  <a
    class="sidebar__link"
    data-v-28ca5493=""
    href="/cadastro-cliente"
  >
    Cadastro de cliente
  </a>
  <a
    class="sidebar__link"
    data-v-28ca5493=""
    href="/cadastro-produto"
  >
    Cadastro de produto
  </a>
  
</div>
`;
